### ISSUE TITLE
Number parsing problem with line gradient

### ISSUE THREAD

**Issue by Lumiadmin** [mapbox/mapbox-maps-android#2256]
## Environment

- Maps SDK Version: v11

## Observed behavior and steps to reproduce

I tried to migrate to v11 from v10 and I encountered a problem. My code sets a line gradient programmatically. It works fine when the code get executed the first time, but on the second attempt it throws an exception:

java.lang.NumberFormatException: For input string: "-1e-8"

I inspected it with the debugger and the first execution indeed sets such a value. However, this does not result in an error. The problem only occurs when reading the layer after this. This is what the json looks like when inspecting with the debugger:

```
{
  "paint": {
    "line-gradient": [
      "interpolate",
      [
        "linear"
      ],
      [
        "line-progress"
      ],
      -1e-8, 
...
```
## Expected behavior

No exception thrown. Layer works with the given line-progress value.

**Comment by kiryldz** [comment]
@Lumiadmin thanks for reporting! We are aware of this issue and this is expected to be fixed in v11.2.0-beta.1. It will be out in 2 weeks.

**Comment by kiryldz** [comment]
Fixed in v11.2.
